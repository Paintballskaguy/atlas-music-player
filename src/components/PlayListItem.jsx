import React from "react";

const PlayListItem = ({ title, artist, length, isActive }) => {
  return (
    <div className={`flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-102 ${
      isActive 
        ? 'bg-gradient-to-r from-custom-red-100 to-custom-yellow-100 dark:from-custom-red-900 dark:to-custom-yellow-900 border-l-6 border-custom-red-500 shadow-lg' 
        : 'hover:bg-custom-teal-50 dark:hover:bg-custom-teal-900/30 hover:shadow-md'
    }`}>
      <div className="flex-1">
        <h3 className={`text-lg font-bold leading-tight transition-colors duration-200 ${
          isActive 
            ? 'text-custom-red-700 dark:text-custom-red-300' 
            : 'text-custom-teal-800 dark:text-custom-teal-100 hover:text-custom-teal-600 dark:hover:text-custom-teal-300'
        }`}>
          {title}
        </h3>
        <p className={`text-base mt-1 transition-colors duration-200 ${
          isActive 
            ? 'text-custom-yellow-700 dark:text-custom-yellow-300' 
            : 'text-custom-blue-600 dark:text-custom-blue-300'
        }`}>
          {artist}
        </p>
      </div>
      <span className={`text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 ${
        isActive 
          ? 'text-custom-red-600 dark:text-custom-red-400' 
          : 'text-custom-teal-600 dark:text-custom-teal-400'
      }`}>
        {length}
      </span>
    </div>
  );
};

export default PlayListItem;