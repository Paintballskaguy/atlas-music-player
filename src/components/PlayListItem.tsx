import React from "react";

interface PlayListItemProps {
  title: string;
  artist: string;
  length: string;
  isActive?: boolean;
}

const PlayListItem: React.FC<PlayListItemProps> = ({ 
  title, 
  artist, 
  length, 
  isActive = false 
}) => {
  // Format duration to HH:MM format
  const formatDuration = (duration: string): string => {
    // If already in HH:MM format, return as is
    if (duration.includes(':')) {
      return duration;
    }
    
    // If in seconds, convert to HH:MM
    const totalSeconds = parseInt(duration);
    if (!isNaN(totalSeconds)) {
      const minutes = Math.floor(totalSeconds / 60);
      const seconds = totalSeconds % 60;
      return `${minutes}:${seconds.toString().padStart(2, '0')}`;
    }
    
    return duration; // Return original if can't parse
  };

  return (
    <div className={`flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-105 ${
      isActive 
        ? 'bg-gradient-to-r from-custom-red-100 to-custom-yellow-100 dark:from-custom-red-900 dark:to-custom-yellow-900 border-l-4 border-custom-red-500 shadow-lg' 
        : 'hover:bg-custom-teal-50 dark:hover:bg-custom-teal-900/30 hover:shadow-md'
    }`}>
      <div className="flex-1">
        <h3 className={`text-lg font-bold leading-tight transition-colors duration-200 ${
          isActive 
            ? 'text-custom-red-700 dark:text-custom-red-300' 
            : 'text-custom-teal-800 dark:text-custom-teal-100 hover:text-custom-teal-600 dark:hover:text-custom-teal-300'
        }`}>
          {title}
        </h3>
        <p className={`text-base mt-1 transition-colors duration-200 ${
          isActive 
            ? 'text-custom-yellow-700 dark:text-custom-yellow-300' 
            : 'text-custom-blue-600 dark:text-custom-blue-300'
        }`}>
          {artist}
        </p>
      </div>
      <span className={`text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 ${
        isActive 
          ? 'text-custom-red-600 dark:text-custom-red-400' 
          : 'text-custom-teal-600 dark:text-custom-teal-400'
      }`}>
        {formatDuration(length)}
      </span>
    </div>
  );
};

export default PlayListItem;