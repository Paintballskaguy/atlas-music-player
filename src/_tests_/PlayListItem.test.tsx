import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import PlayListItem from '../components/PlayListItem'

interface PlayListItemProps {
  title: string
  artist: string
  length: string | number
  isActive?: boolean
}

describe('PlayListItem', () => {
  const defaultProps: PlayListItemProps = {
    title: 'Test Song',
    artist: 'Test Artist',
    length: '3:00',
    isActive: false,
  }

  it('renders with default props', () => {
    const { container } = render(<PlayListItem {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders as active song', () => {
    const props: PlayListItemProps = {
      ...defaultProps,
      isActive: true
    }
    
    const { container } = render(<PlayListItem {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with different song data', () => {
    const props: PlayListItemProps = {
      title: 'Another Song',
      artist: 'Different Artist',
      length: '4:30',
      isActive: false
    }
    
    const { container } = render(<PlayListItem {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with numeric length (seconds)', () => {
    const props: PlayListItemProps = {
      ...defaultProps,
      length: 125 // 2 minutes 5 seconds
    }
    
    const { container } = render(<PlayListItem {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with long title and artist', () => {
    const props: PlayListItemProps = {
      title: 'Very Long Song Title That Might Wrap Around The Container',
      artist: 'Very Long Artist Name That Might Also Wrap Around',
      length: '5:45',
      isActive: false
    }
    
    const { container } = render(<PlayListItem {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('formats duration correctly for different inputs', () => {
    // Test various duration formats
    const testCases = [
      { input: '3:30', expected: '3:30' }, // Already formatted
      { input: 125, expected: '2:05' },    // Seconds to MM:SS
      { input: '125', expected: '2:05' },  // String seconds to MM:SS
      { input: 'invalid', expected: 'invalid' } // Fallback to original
    ]

    testCases.forEach(({ input, expected }) => {
      const props: PlayListItemProps = {
        title: 'Test Song',
        artist: 'Test Artist',
        length: input,
        isActive: false
      }
      
      const { container } = render(<PlayListItem {...props} />)
      const durationElement = container.querySelector('span')
      expect(durationElement?.textContent).toBe(expected)
    })
  })

  it('applies active styling when isActive is true', () => {
    const props: PlayListItemProps = {
      ...defaultProps,
      isActive: true
    }
    
    const { container } = render(<PlayListItem {...props} />)
    const listItem = container.firstChild as HTMLElement
    
    // Check for active styling classes
    expect(listItem.className).toContain('bg-gradient-to-r')
    expect(listItem.className).toContain('border-l-4')
    expect(listItem.className).toContain('border-custom-red-500')
  })

  it('applies hover styling when not active', () => {
    const props: PlayListItemProps = {
      ...defaultProps,
      isActive: false
    }
    
    const { container } = render(<PlayListItem {...props} />)
    const listItem = container.firstChild as HTMLElement
    
    // Check for hover styling classes
    expect(listItem.className).toContain('hover:bg-custom-teal-50')
    expect(listItem.className).toContain('hover:shadow-md')
  })
})