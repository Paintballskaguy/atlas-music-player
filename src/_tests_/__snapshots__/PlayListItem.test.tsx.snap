// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem > renders as active song 1`] = `
<div
  class="flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-105 bg-gradient-to-r from-custom-red-100 to-custom-yellow-100 dark:from-custom-red-900 dark:to-custom-yellow-900 border-l-4 border-custom-red-500 shadow-lg"
>
  <div
    class="flex-1"
  >
    <h3
      class="text-lg font-bold leading-tight transition-colors duration-200 text-custom-red-700 dark:text-custom-red-300"
    >
      Test Song
    </h3>
    <p
      class="text-base mt-1 transition-colors duration-200 text-custom-yellow-700 dark:text-custom-yellow-300"
    >
      Test Artist
    </p>
  </div>
  <span
    class="text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 text-custom-red-600 dark:text-custom-red-400"
  >
    3:00
  </span>
</div>
`;

exports[`PlayListItem > renders with default props 1`] = `
<div
  class="flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-105 hover:bg-custom-teal-50 dark:hover:bg-custom-teal-900/30 hover:shadow-md"
>
  <div
    class="flex-1"
  >
    <h3
      class="text-lg font-bold leading-tight transition-colors duration-200 text-custom-teal-800 dark:text-custom-teal-100 hover:text-custom-teal-600 dark:hover:text-custom-teal-300"
    >
      Test Song
    </h3>
    <p
      class="text-base mt-1 transition-colors duration-200 text-custom-blue-600 dark:text-custom-blue-300"
    >
      Test Artist
    </p>
  </div>
  <span
    class="text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 text-custom-teal-600 dark:text-custom-teal-400"
  >
    3:00
  </span>
</div>
`;

exports[`PlayListItem > renders with different song data 1`] = `
<div
  class="flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-105 hover:bg-custom-teal-50 dark:hover:bg-custom-teal-900/30 hover:shadow-md"
>
  <div
    class="flex-1"
  >
    <h3
      class="text-lg font-bold leading-tight transition-colors duration-200 text-custom-teal-800 dark:text-custom-teal-100 hover:text-custom-teal-600 dark:hover:text-custom-teal-300"
    >
      Another Song
    </h3>
    <p
      class="text-base mt-1 transition-colors duration-200 text-custom-blue-600 dark:text-custom-blue-300"
    >
      Different Artist
    </p>
  </div>
  <span
    class="text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 text-custom-teal-600 dark:text-custom-teal-400"
  >
    4:30
  </span>
</div>
`;

exports[`PlayListItem > renders with long title and artist 1`] = `
<div
  class="flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-105 hover:bg-custom-teal-50 dark:hover:bg-custom-teal-900/30 hover:shadow-md"
>
  <div
    class="flex-1"
  >
    <h3
      class="text-lg font-bold leading-tight transition-colors duration-200 text-custom-teal-800 dark:text-custom-teal-100 hover:text-custom-teal-600 dark:hover:text-custom-teal-300"
    >
      Very Long Song Title That Might Wrap Around The Container
    </h3>
    <p
      class="text-base mt-1 transition-colors duration-200 text-custom-blue-600 dark:text-custom-blue-300"
    >
      Very Long Artist Name That Might Also Wrap Around
    </p>
  </div>
  <span
    class="text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 text-custom-teal-600 dark:text-custom-teal-400"
  >
    5:45
  </span>
</div>
`;

exports[`PlayListItem > renders with numeric length (seconds) 1`] = `
<div
  class="flex items-start justify-between py-4 px-4 cursor-pointer rounded-2xl transition-all duration-300 transform hover:scale-105 hover:bg-custom-teal-50 dark:hover:bg-custom-teal-900/30 hover:shadow-md"
>
  <div
    class="flex-1"
  >
    <h3
      class="text-lg font-bold leading-tight transition-colors duration-200 text-custom-teal-800 dark:text-custom-teal-100 hover:text-custom-teal-600 dark:hover:text-custom-teal-300"
    >
      Test Song
    </h3>
    <p
      class="text-base mt-1 transition-colors duration-200 text-custom-blue-600 dark:text-custom-blue-300"
    >
      Test Artist
    </p>
  </div>
  <span
    class="text-base ml-4 flex-shrink-0 font-semibold transition-colors duration-200 text-custom-teal-600 dark:text-custom-teal-400"
  >
    2:05
  </span>
</div>
`;
